update the spec for group/add-member to be v2!

mixmix opened the pull request:
We have a required change to the format of the `group/add-member` message:
- the feedId in recps MUST be the root feed, which is format bendybutt!
- the version needs to be bumped (this is not the v1 schema any more!)

If we're making breaking changes lets change `groupKey` to `secret`

Look at us, version out specs!

:fire: this PR will require us to do some changes to schema in this spec. Will wait for other PRs to settle down. We may also need to keep both version of the spec, i.e. v1 and v2, as if there are json files being pulled in by Ahau for migration

Powersource reviewed (changes requested):
Makes sense. Just need to update the schemas now.

Powersource commented:
Also yeah maybe different conversation but how will ahau do things? Stay on v1 and have 1.1, 1.2 etc?

mixmix commented:
@Powersource ready for review!!!

Changes made:
- stored an old version of the v1 `group/add-member` schema (Ahau will need v1 + v2)
- modified the `group/add-member` schema
  - added a `creator` field (in case the person receiving the add-member doesn't replicate the group creator yet, which is needed)
  - changed `groupKey` > `secret` (I like this term more now... why not... future mix will hate me I'm sure)
- all specs - changes the FeedId definition to be `bendybutt-v1` (this is the only id used in recps now)

mixmix commented:
decided to undo the `groupKey` to `secret` change. I wish it was  `secret`, but it will just cause a cascade of code changes elsewhere which is silly

Powersource commented on group/add-member/v2/schema.js:
```diff
@@ -35,7 +36,7 @@ module.exports = {
       properties: {
         group: { $ref: '#/definitions/tangle/update' },
-        members: { $ref: '#/definitions/tangle/any' }
+        members: { $ref: '#/definitions/tangle/update' }
```
shouldn't this have an explicit root somewhere though? related https://github.com/ssbc/ssb-tribes2/issues/33

mixmix replied:
It could, currently it's hanging off the group/init which I think is fine.
It all depends on how we define our tangle, and how we define the "reduce" of the info in that tangle.

Gonna merge this for the moment @Powersource

Powersource commented:
@mixmix let's get this merged and published :) would be nice to have addMember verification back in tribes2

Powersource commented:
@mixmix boop

mixmix commented:
Pubished as `v3.0.0`
